### ISSUE TITLE
Editor crashes when MonsterShield becomes connected

### ISSUE THREAD

**Issue by kacole2** [Xpendable172/MonsterShield#1]
Hey Jason! How's it going? It's been a while but was hoping you could answer this whenever you get around to it. If you don't recall, I was the one who created the [MonsterRemote](https://github.com/kacole2/monster_remote) based on your genius here. I'm sorry that this project didn't go much further. It had such great potential. I wish I could have taken it over for you but I ended up selling the house that hosted my haunt and moved. I gifted the MonserShield and the bulk of my yard haunt to my neighbors to continue the tradition. 

I was over there tonight and I ran into all kinds of problems (including this issue) but was able to work through most of it. The fact that they lost their animation file made it *that* much harder!

Anyway, I know there has to be something I'm overlooking but it's happening on 2 different laptops, one that includes my XP Virtual Machine. I can fire up the MonsterShield Editor, edit tracks, add music, change animations, etc. However, whenever I hit the `connect` button to talk through the COM/USB interface, the Editor program crashes. The weird part is that I can move the crash window over and continue working on the editor, however it's very laggy and can't be certain it's working correctly. I was successfully able to change 1 of the animations that needed to be done. After some more troubleshooting, we got the MonsterShield with MonsterRemote up and running  and setting off props again! 

At this point, they aren't in a good situation to ever create new tracks but wanted to see if this crash is related to some driver needed or Visual Studio dll that's missing. I ran this on both XP and Windows 10 and the same error occurred. This is not a critical issue because they don't plan on changing any animations this year but wanted to submit this bug in case anyone else is having the issue. 

Hope all is well
Kenny

**Comment by Xpendable172** [comment]
Yes, I of course remember you and the awesome MonsterRemote you had
created, as well as your Halloween light show. As you obviously know, I am
no longer involved with the MonsterShield. I am probably not going to be of
much help, but here is what I know... The serial communications (through
the USB / COM) interface can be touchy with the editor. Make sure you
select the right com port. If you select the wrong one I can guarantee it
will likely crash. If your cable is at all flaky or if Windows removes the
com port, you will likely crash. I did experience random crashes out in the
field when working on the Pumpkin Farm and would have to unplug the USB
cable, reset the MonsterShield (not a factory reset, just a quick tap on
reset button), then plug it in then start the editor. Make sure you give
enough time for the com port to reappear. Always always always save your
sequences before attempting to send them. Also recommend saving into
different filenames by adding a version number so you can go back if
something goes wrong along the way. Good luck. Sorry I can't be more help.

Sincerely,
Jason

**Comment by kacole2** [comment]
@Xpendable172 thanks for the quick response! Two machines, same issue. So it's not the COM port because the window shows serial input/output happening during the connection phase. I'll have to try the cable unplug/tap reset/plug-in next time I'm around. 

I figured you weren't maintaining any more but wanted to let you know your creation is still alive and kicking on the biggest halloween street in Louisville, KY.

After sending this message I've spent the entire time seeing if anyone has created a similar board/software combo that has feature parity for the cost... guess what? nothing! Looks like a lot of people have gone the route of Christmas light controllers. I've spent 15 minutes reading up on Falcon and still haven't been able to tell if it could give me what I would have needed. I don't even do crazy halloween stuff anymore but just got sucked into it again. I really miss it during this time of the year. Take care!
